hyperparameter_tuning.py

```diff
@@ -0,0 +1,155 @@
+import gc
+
+import numpy as np
+from deepsense import neptune
+from sklearn.externals import joblib
+
+from steps.base import BaseTransformer
+from utils import set_seed
+
+
+class RandomSearchOptimizer(BaseTransformer):
```

Comment:
add docstring to this class. One have no idea what are these params.
Use Type hints.